src/dividat-driver/firmware/main.go

```diff
@@ -129,9 +129,9 @@ func Update(parentCtx context.Context, image io.Reader, deviceSerial *string, co
 		}
 	}
 
-	onProgress("Waiting 5 seconds to ensure proper TFTP startup")
+	onProgress("Waiting 15 seconds to ensure proper TFTP startup")
 	// 4: Transmit firmware via TFTP
-	time.Sleep(5 * time.Second) // Wait to ensure proper TFTP startup
```

Comment:
Another thing we could do, if we want to optimize for time, is to add the extra wait only in the case where we use a `configuredAddr` for the `dfuHost`. Then we don't need to do extra waiting in the mDNS scenario.

Reply:
In [Refactor discovery code](https://github.com/dividat/driver/pull/127/commits/c7608d7ff76ea1cdbb47d48a1f56cf3597b9122a) I added support for getting the `mode=` txt entry. 

https://github.com/dividat/driver/blob/6da5b8bb37abf2a9b5d1d3d4c68e6393c4d0d4aa/src/dividat-driver/service/main.go#L142-L155

The `service.Find` call here spawns two goroutines looking for both `_sensoControl._tcp` and `_sensoUpdate._udp` and returns whichever is found first:
https://github.com/dividat/driver/blob/6da5b8bb37abf2a9b5d1d3d4c68e6393c4d0d4aa/src/dividat-driver/firmware/cli.go#L53

With this we can run a discovery first even for a configured address which allows us to learn right away if a Senso is in normal or bootloader mode (as well as validating the address). Then this timeout can just be part of the dfu sending logic:
https://github.com/dividat/driver/blob/6da5b8bb37abf2a9b5d1d3d4c68e6393c4d0d4aa/src/dividat-driver/firmware/main.go#L38-L68